types/frida-gum/index.d.ts

```diff
@@ -0,0 +1,6668 @@
+}
+
+declare const enum MemoryOperation {
+    read = "read",
+    write = "write",
+    execute = "execute"
+}
+
+declare interface EnumerateCallbacks<T> {
+    onMatch: (item: T) => void | EnumerateAction;
+    onComplete: () => void;
+}
+
+declare const enum EnumerateAction {
+    Stop = "stop"
+}
+
+declare interface MemoryScanCallbacks {
+    /**
+     * Called with each occurence that was found.
+     *
+     * @param address Memory address where a match was found.
+     * @param size Size of this match.
+     */
+    onMatch: (address: NativePointer, size: number) => void | EnumerateAction;
+
+    /**
+     * Called when there was a memory access error while scanning.
+     *
+     * @param reason Why the memory access failed.
+     */
+    onError?: (reason: string) => void;
+
+    /**
+     * Called when the memory range has been fully scanned.
+     */
+    onComplete: () => void;
+}
+
+declare interface MemoryScanMatch {
+    /**
+     * Memory address where a match was found.
+     */
+    address: NativePointer;
+
+    /**
+     * Size of this match.
+     */
+    size: number;
+}
+
+declare interface KernelMemoryScanCallbacks {
+    /**
+     * Called with each occurence that was found.
+     *
+     * @param address Memory address where a match was found.
+     * @param size Size of this match.
+     */
+    onMatch: (address: UInt64, size: number) => void | EnumerateAction;
+
+    /**
+     * Called when there was a memory access error while scanning.
+     *
+     * @param reason Why the memory access failed.
+     */
+    onError?: (reason: string) => void;
+
+    /**
+     * Called when the memory range has been fully scanned.
+     */
+    onComplete: () => void;
+}
+
+declare interface KernelMemoryScanMatch {
+    /**
+     * Memory address where a match was found.
+     */
+    address: UInt64;
+
+    /**
+     * Size of this match.
+     */
+    size: number;
+}
+
+declare type MemoryPatchApplyCallback = (code: NativePointer) => void;
+
+/**
+ * Represents a signed 64-bit value.
+ */
+declare class Int64 {
+    /**
+     * Creates a new Int64 from `v`, which is either a string containing the value in decimal, or hexadecimal
+     * if prefixed with “0x”, or a number. You may use the int64(v) short-hand for brevity.
+     */
+    constructor(v: string | number | Int64);
+
+    /**
+     * Makes a new Int64 whose value is `this` + `v`.
+     */
+    add(v: Int64 | number | string): Int64;
+
+    /**
+     * Makes a new Int64 whose value is `this` - `v`.
+     */
+    sub(v: Int64 | number | string): Int64;
+
+    /**
+     * Makes a new Int64 whose value is `this` & `v`.
+     */
+    and(v: Int64 | number | string): Int64;
+
+    /**
+     * Makes a new Int64 whose value is `this` | `v`.
+     */
+    or(v: Int64 | number | string): Int64;
+
+    /**
+     * Makes a new Int64 whose value is `this` ^ `v`.
+     */
+    xor(v: Int64 | number | string): Int64;
+
+    /**
+     * Makes a new Int64 whose value is `this` << `v`.
```

Comment:
I admit I don't use it very often, but isn't `<<` left-shift?

Reply:
Oops, indeed, the docstrings somehow ended up reversed. 😅